src/v/finjector/stress_fiber.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "finjector/stress_fiber.h"
+
+#include "random/generators.h"
+#include "ssx/future-util.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/timer.hh>
+
+class stress_payload {
+public:
+    // Starts the stress payload with the given configuration.
+    static std::unique_ptr<stress_payload> start(stress_config cfg) {
+        return std::unique_ptr<stress_payload>(new stress_payload(cfg));
```

Comment:
This doesn't work with private constructors since make_unique would need to be a friend. Lmk if you feel strongly about this; otherwise I kind of prefer keeping the constructor private stylistically. IMO it makes it clearer that constructing actually starts the payload.

Reply:
oh right, i didn't notice it was private. sgtm